### PR TITLE
Frank

### PR DESCRIPTION
**Pull request by fshi7418**
if possible, only add pollutants_analysis1.R

### CODE REVIEW

File: pollutants_analysis1.R [hunk 1]
```diff
@@ -0,0 +1,108 @@
+set.seed(123) ## for reproducibility
+library(glmnet)
+library(corrplot)
+data_dir <-  r"(C:\Users\Frank Shi\Documents\FrankS\Waterloo\pollutants)"
+setwd(data_dir)
+pollutants_original <- read.csv(file='pollutants.csv')
+
+#===============================================
+# plot correlation bewtween all covariates
+excl_cols <- c('X', 'length')
```
Comment (RootofalleviI) [inline pollutants_analysis1.R:10]: What's this `X` for?

File: pollutants_analysis1.R [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+pollutants1 <- pollutants_original[!colnames(pollutants_original) %in% excl_cols]
+pollutants_corr <- cor(pollutants1)
+corrplot(pollutants_corr, order="hclust", method="color", type="full",
```
Comment (j32duan) [inline pollutants_analysis1.R:13]: todo1: prettify this plot if we need to use it in the final report.

File: pollutants_analysis1.R [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+#===============================================
+# plot correlation bewtween all covariates
+excl_cols <- c('X', 'length')
+pollutants1 <- pollutants_original[!colnames(pollutants_original) %in% excl_cols]
+pollutants_corr <- cor(pollutants1)
+corrplot(pollutants_corr, order="hclust", method="color", type="full", 
+         tl.col = "black")
+
+# take out a few covariates based on the correlation heat map
+excl_cols1 <- c('X', paste0('POP_PCB', c(1, 2, 4, 5, 7, 9)),
```
Comment (j32duan) [inline pollutants_analysis1.R:17]: What is this `X` for?

**Comment by RootofalleviI** [comment]
Good stuff. We should now evaluate the models selected by these methods.

File: pollutants_analysis1.R [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+# pollutants <- pollutants[!colnames(pollutants) %in% adhoc_excl]
+
+#===============================================
+# establishing benchmarks
+model_full <- lm(length ~ ., data=pollutants)
+model_0 <- lm(length ~ 1, data=pollutants)
+model_start_aic <- lm(length ~ 1, data=pollutants)
+# aic stepwise
+system.time({
+  Mstep <- step(object = model_start_aic,
```
Comment (j32duan) [inline pollutants_analysis1.R]: Let's rename this to Mstep_aic?

File: pollutants_analysis1.R [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                scope = list(lower = model_0, upper = model_full),
+                direction = "both", 
+                trace = 1, 
+                k = 2)
+})
+
+# bic stepwise
+model_start_bic <- lm(length ~ 1, data=pollutants)
+system.time({
+  Mstep <- step(object = model_start_bic,
```
Comment (j32duan) [inline pollutants_analysis1.R]: Let's rename this to Mstep_bic?

**Review by j32duan — CHANGES_REQUESTED** [review]

**Comment by j32duan** [comment]
Good stuff. We should now start evaluating these models.
Also be careful with the order of train-test-split/cross-validation and the feature selection process.

File: pollutants_analysis1.R [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                trace = 1, 
+                k = log(nrow(pollutants)))
+})
+
+#  lasso
+y <- pollutants[['length']]
+X <- model.matrix(model_full)
+model_lasso <- glmnet(x=X, y=y, alpha=1, nlambda=500)
+cv_fit_lasso <- cv.glmnet(x=X, y=y, alpha=1, nlambda=500)
+print(coef(model_lasso, s="lambda.min"))
```
Comment (loganwatchorn) [inline pollutants_analysis1.R:93]: Fails to print for me

File: pollutants_analysis1.R (hunk 3 above)
Comment (loganwatchorn) [reply pollutants_analysis1.R:17]: X is the column indexing row numbers, included in the original dataset

File: pollutants_analysis1.R (hunk 6 above)
Comment (loganwatchorn) [reply pollutants_analysis1.R:93]: But since it works for Frank let's ignore this.

**Review by loganwatchorn — APPROVED** [review]

**Pull request merged**